src/Keboola/DbWriter/Adapter/AbsAdapter.php

```diff
@@ -63,42 +67,82 @@ public function generateCreateStageCommand(string $stageName): string
-                if (!empty($column['nullable'])) {
-                    return sprintf("IFF(t.$%d = '', null, t.$%d)", $index + 1, $index + 1);
-                }
-                return sprintf('t.$%d', $index + 1);
-            },
-            $columns,
-            array_keys($columns)
-        );
+        $filesToImport = $this->getManifestEntries();
+        foreach (array_chunk($filesToImport, self::SLICED_FILES_CHUNK_SIZE) as $files) {
+            $quotedFiles = array_map(
+                function ($entry) {
+                    return Snowflake::quote(
+                        strtr($entry, [$this->getContainerUrl() . '/' => ''])
+                    );
+                },
+                $files
+            );
+
+            yield sprintf(
+                'COPY INTO %s(%s) 
+            FROM (SELECT %s FROM %s t)
+            FILES = (%s)',
+                $tableName,
+                implode(', ', SqlHelper::getQuotedColumnsNames($columns)),
+                implode(', ', SqlHelper::getColumnsTransformation($columns)),
+                Snowflake::quote('@' . Snowflake::quoteIdentifier($stageName) . '/'),
+                implode(',', $quotedFiles)
+            );
+        }
+    }
 
-        $path = $this->name;
-        $pattern = '';
-        if ($this->isSliced) {
-            // key ends with manifest
-            if (strrpos($this->name, 'manifest') === strlen($this->name) - strlen('manifest')) {
-                $path = substr($this->name, 0, strlen($this->name) - strlen('manifest'));
-                $pattern = 'PATTERN="^.*(?<!manifest)$"';
+    private function getManifestEntries(): array
+    {
+        $blobClient = $this->getClient();
+        if (!$this->isSliced) {
+            // this is temporary solution copy into is not failing when blob not exists
+            try {
+                $blobClient->getBlob($this->container, $this->name);
+            } catch (ServiceException $e) {
+                throw new UserException('Load error: ' . $e->getErrorText(), 0, $e);
             }
+
+            [$this->getContainerUrl() . $this->name];
         }
 
-        return sprintf(
-            'COPY INTO %s(%s) 
-            FROM (SELECT %s FROM %s t)
-            %s',
-            $tableName,
-            implode(', ', $columnNames),
-            implode(', ', $transformationColumns),
-            Snowflake::quote('@' . Snowflake::quoteIdentifier($stageName) . '/' . $path),
-            $pattern
+        try {
+            $manifestBlob = $blobClient->getBlob($this->container, $this->name);
+        } catch (ServiceException $e) {
+            throw new UserException('Load error: manifest file was not found.', 0, $e);
+        }
+
+        $manifest = \GuzzleHttp\json_decode((string) stream_get_contents($manifestBlob->getContentStream()), true);
+        return array_map(function (array $entry) use ($blobClient) {
+            // this is temporary solution copy into is not failing when blob not exists
```

Comment:
tady to ve skutečnosti pro SNFLK není potřeba jen pro synapse, protože ten nefailne když některý soubor neexistuje snflk jo.

Reply:
vdaka. to mi nedoslo.

Reply:
https://github.com/keboola/db-writer-snowflake/pull/82/commits/8ba953777473e31b8b3b2c981b81ea2962e76583